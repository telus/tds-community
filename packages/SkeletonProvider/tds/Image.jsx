import React from 'react'
import PropTypes from 'prop-types'
import Image from '@tds/core-image'
import withSkeleton from '../withSkeleton'
import styles from '../Skeleton.scss'

const ImageSkeleton = ({ skeleton, ...rest }) => {
  if (!skeleton) {
    return <Image {...rest} />
  }

  const baseClass = styles['skeleton-image']
  const circularClass = styles.circular

  const cssWidth = `${rest.width * 0.8}px`
  const cssHeight = `${rest.height * 0.8}px`

  const classes = [baseClass, circularClass].join(' ')

  return <div className={classes} style={{ width: cssWidth, height: cssHeight }} />
}

ImageSkeleton.propTypes = {
  skeleton: PropTypes.bool,
}

ImageSkeleton.defaultProps = {
  skeleton: false,
}

export default withSkeleton(ImageSkeleton)(Image)
