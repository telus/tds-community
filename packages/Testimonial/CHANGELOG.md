# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [3.1.0](https://github.com/telus/tds-community/compare/@tds/community-testimonial@3.0.5...@tds/community-testimonial@3.1.0) (2021-09-21)


### Features

* **community-testimonial:** react 17 and styled components 5 ([123ae82](https://github.com/telus/tds-community/commit/123ae82ac86b6f636819091e5a5ed2e8157b512d))





## [3.0.5](https://github.com/telus/tds-community/compare/@tds/community-testimonial@3.0.4...@tds/community-testimonial@3.0.5) (2020-09-09)

**Note:** Version bump only for package @tds/community-testimonial





## [3.0.4](https://github.com/telus/tds-community/compare/@tds/community-testimonial@3.0.3...@tds/community-testimonial@3.0.4) (2020-09-09)


### Bug Fixes

* **community-testimonial:** update util-helpers version ([10b8c5a](https://github.com/telus/tds-community/commit/10b8c5a3461cc916f1a90799eb7ef593835b7421))





## [3.0.3](https://github.com/telus/tds-community/compare/@tds/community-testimonial@3.0.2...@tds/community-testimonial@3.0.3) (2020-04-13)

**Note:** Version bump only for package @tds/community-testimonial





## [3.0.2](https://github.com/telus/tds-community/compare/@tds/community-testimonial@3.0.1...@tds/community-testimonial@3.0.2) (2020-03-19)

**Note:** Version bump only for package @tds/community-testimonial





## [3.0.1](https://github.com/telus/tds-community/compare/@tds/community-testimonial@3.0.0...@tds/community-testimonial@3.0.1) (2019-09-17)

**Note:** Version bump only for package @tds/community-testimonial





# [3.0.0](https://github.com/telus/tds-community/compare/@tds/community-testimonial@2.1.0...@tds/community-testimonial@3.0.0) (2019-08-29)


### Features

* **community-testimonial:** upgrade to styled-components ([518c30b](https://github.com/telus/tds-community/commit/518c30b))


### BREAKING CHANGES

* **community-testimonial:** styled-components now a peer dependency





# [2.1.0](https://github.com/telus/tds-community/compare/@tds/community-testimonial@2.0.0...@tds/community-testimonial@2.1.0) (2019-08-26)


### Features

* **community-testimonial:** add testimonialStyle prop ([d1b5eb2](https://github.com/telus/tds-community/commit/d1b5eb2))





# [2.0.0](https://github.com/telus/tds-community/compare/@tds/community-testimonial@1.0.5...@tds/community-testimonial@2.0.0) (2019-05-22)


### Bug Fixes

* **community-testimonial:** changes paragraph size on body text ([9c9ecb8](https://github.com/telus/tds-community/commit/9c9ecb8))


### Features

* **community-testimonial:** remove card from community-testimonial ([48158ca](https://github.com/telus/tds-community/commit/48158ca)), closes [#192](https://github.com/telus/tds-community/issues/192)


### BREAKING CHANGES

* **community-testimonial:** removes card from community-testimonial component





## [1.0.5](https://github.com/telus/tds-community/compare/@tds/community-testimonial@1.0.4...@tds/community-testimonial@1.0.5) (2019-05-17)

**Note:** Version bump only for package @tds/community-testimonial





## [1.0.4](https://github.com/telus/tds-community/compare/@tds/community-testimonial@1.0.3...@tds/community-testimonial@1.0.4) (2019-03-27)

**Note:** Version bump only for package @tds/community-testimonial





## [1.0.3](https://github.com/telus/tds-community/compare/@tds/community-testimonial@1.0.2...@tds/community-testimonial@1.0.3) (2019-01-08)

### Bug Fixes

- **community-testimonial:** bump version of [@tds](https://github.com/tds)/util-prop-types ([8e934e0](https://github.com/telus/tds-community/commit/8e934e0))

<a name="1.0.2"></a>

## [1.0.2](https://github.com/telus/tds-community/compare/@tds/community-testimonial@1.0.1...@tds/community-testimonial@1.0.2) (2018-11-27)

### Bug Fixes

- **community-testimonial:** move util-prop-types to dependencies ([1e3ccf5](https://github.com/telus/tds-community/commit/1e3ccf5))
- **community-testimonial:** refactor proptypes package ([ec2efbd](https://github.com/telus/tds-community/commit/ec2efbd))

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telus/tds-community/compare/@tds/community-testimonial@1.0.0...@tds/community-testimonial@1.0.1) (2018-10-22)

**Note:** Version bump only for package @tds/community-testimonial

<a name="1.0.0"></a>

# 1.0.0 (2018-10-10)

### Bug Fixes

- **community-testimonial:** add defaultProps ([98a314b](https://github.com/telus/tds-community/commit/98a314b))
- **community-testimonial:** add responsive styles ([10ff101](https://github.com/telus/tds-community/commit/10ff101))

### Features

- **community-testimonial:** added testimonial component ([be4dc67](https://github.com/telus/tds-community/commit/be4dc67))
- **community-testimonial:** component refactor based on PR comments ([bc6e89c](https://github.com/telus/tds-community/commit/bc6e89c))
