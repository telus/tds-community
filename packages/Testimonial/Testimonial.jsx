import React from 'react'
import PropTypes from 'prop-types'

import Box from '@tds/core-box'
import Paragraph from '@tds/core-paragraph'
import safeRest from '@tds/shared-safe-rest'
import { componentWithName } from '@tds/util-prop-types'

import styles from './Testimonial.scss'

/**
 * Testimonial component for displaying testimonial in a standalone, pre-styled component.
 * @version ./package.json
 */
const Testimonial = ({ testimonial, image, title, additionalInfo, ...rest }) => {
  return (
    <Box {...safeRest(rest)} between={3}>
      <Paragraph size="large">{testimonial}</Paragraph>
      <div className={styles.imageAndQuote}>
        {image && <div className={styles.image}>{image}</div>}
        <div>
          <Paragraph bold>{title}</Paragraph>
          <Paragraph>{additionalInfo}</Paragraph>
        </div>
      </div>
    </Box>
  )
}

Testimonial.propTypes = {
  /** Testimonial main text */
  testimonial: PropTypes.string.isRequired,
  /** Image Component - suggested & max size is 60x60px. */
  image: componentWithName('Image'),
  /** Image alt text - needed for accessibility */
  title: PropTypes.string.isRequired,
  /** Testimonial author credentials - more info - normal weight */
  additionalInfo: PropTypes.string.isRequired,
}

Testimonial.defaultProps = {
  image: undefined,
}

export default Testimonial
