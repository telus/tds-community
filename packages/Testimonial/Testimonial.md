```
<Testimonial
  testimonial={`“My office is where ever my customers are. TELUS helps me stay connected whether I'm making a sale or doing payroll.”`}
  image={<Image
  src="image-example.jpg"
  rounded="circle"
  width={60}
  height={60}
  alt="Image of co-workers collaborating"
  />}
  title="Dave Smith, Foreman"
  additionalInfo="Pinnacle Reforestation"
/>
```
