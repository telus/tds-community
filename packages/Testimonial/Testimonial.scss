@import '~@tds/core-responsive/responsive';

.imageAndQuote {
  display: flex;
  flex-direction: column;

  @include mq($from: sm) {
    flex-direction: row;
    align-items: center;
  }
}

.image {
  font-size: 0;
  margin-bottom: 1rem;
  max-width: 60px;
  max-height: 60px;

  @include mq($from: sm) {
    display: flex;
    align-items: center;
    max-height: 60px;
    max-width: 60px;
    margin-right: 1rem;
    margin-bottom: 0;
  }
}
