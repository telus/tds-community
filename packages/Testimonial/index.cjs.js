require('./dist/index.css')
const Testimonial = require('./dist/index.cjs')

module.exports = Testimonial
