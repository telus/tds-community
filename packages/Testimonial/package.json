{
  "name": "@tds/community-testimonial",
  "version": "1.0.0-preview",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telus/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "@tds/core-box": "^1.0.1",
    "@tds/core-card": "^1.0.1",
    "@tds/core-paragraph": "^1.0.2",
    "@tds/core-responsive": "^1.1.0",
    "prop-types": "^15.6.0"
  },
  "devDependencies": {
    "@tds/shared-safe-rest": "^1.0.0",
    "airbnb-prop-types": "^2.11.0"
  }
}
