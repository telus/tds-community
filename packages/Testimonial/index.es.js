import './dist/index.css'
import Testimonial from './dist/index.es'

export default Testimonial
