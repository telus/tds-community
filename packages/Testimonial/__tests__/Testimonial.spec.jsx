import React from 'react'
import { shallow } from 'enzyme'

import Image from '@tds/core-image'

import Testimonial from '../Testimonial'

describe('Testimonial', () => {
  const doShallow = (props = {}) => shallow(<Testimonial {...props} />)

  const props = {
    testimonial:
      "My office is where ever my customers are. TELUS helps me stay connected whether I'm making a sale or doing payroll.",
    image: (
      <Image
        src="image-example.jpg"
        rounded="circle"
        width={60}
        height={60}
        alt="Image of co-workers collaborating"
      />
    ),
    title: 'Dave Smith, Foreman',
    additionalInfo: 'Pinnacle Reforestation',
  }

  it('renders', () => {
    const testimonial = doShallow({ ...props })

    expect(testimonial).toMatchSnapshot()
  })

  it('renders without an image', () => {
    const testimonial = doShallow({ ...props, image: null })
    expect(testimonial.find('.image').length).toEqual(0)
  })

  it('renders with an image', () => {
    const testimonial = doShallow({ ...props })
    expect(testimonial.find('.image').exists()).toEqual(true)
  })

  it('passes additional attributes to the element', () => {
    const testimonial = doShallow({ id: 'the-id', 'data-some-attr': 'some value', ...props })

    expect(testimonial).toHaveProp('id', 'the-id')
    expect(testimonial).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const testimonial = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
      ...props,
    })

    expect(testimonial).not.toHaveProp('className', 'my-custom-class')
    expect(testimonial).not.toHaveProp('style')
  })
})
