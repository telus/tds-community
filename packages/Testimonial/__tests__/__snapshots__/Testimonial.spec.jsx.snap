// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonial renders 1`] = `
<Box
  between={3}
  inline={false}
  tag="div"
>
  <Paragraph
    align="left"
    bold={false}
    invert={false}
    size="large"
  >
    My office is where ever my customers are. TELUS helps me stay connected whether I'm making a sale or doing payroll.
  </Paragraph>
  <div
    className="imageAndQuote"
  >
    <div
      className="image"
    >
      <Image
        alt="Image of co-workers collaborating"
        height={60}
        rounded="circle"
        src="image-example.jpg"
        width={60}
      />
    </div>
    <div>
      <Paragraph
        align="left"
        bold={true}
        invert={false}
        size="medium"
      >
        Dave Smith, Foreman
      </Paragraph>
      <Paragraph
        align="left"
        bold={false}
        invert={false}
        size="medium"
      >
        Pinnacle Reforestation
      </Paragraph>
    </div>
  </div>
</Box>
`;
